server/reviewStatCalculations.js

```diff
@@ -0,0 +1,49 @@
+const calculateRatingGroup = (percentPositive) => {
+  let ratingGroup = '';
+
+  if (percentPositive > 0.8 && percentPositive <= 1) {
+    ratingGroup = 'Very Positive';
+  } else if (percentPositive > 0.6 && percentPositive < 0.8) {
+    ratingGroup = 'Mostly Positive';
+  } else if (percentPositive > 0.4 && percentPositive < 0.6) {
+    ratingGroup = 'Mixed';
+  } else if (percentPositive > 0.2 && percentPositive < 0.4) {
+    ratingGroup = 'Mostly Negative';
+  } else if (percentPositive >= 0 && percentPositive < 0.2) {
+    ratingGroup = 'Very Negative';
+  }
+
+  return ratingGroup;
+};
+
+const calculateReviewStats = (initialReviewStats) => {
+  const reviewStats = {};
+
+  const hasOverallReviews = initialReviewStats.totalReviewCount > 0;
+  const hasRecentReviews = initialReviewStats.totalRecentReviewCount > 0;
+
+  let percentPositiveOverall;
+  let percentPositiveRecent;
+
+  hasOverallReviews ? percentPositiveOverall = (initialReviewStats.totalPositiveReviewCount / initialReviewStats.totalReviewCount).toFixed(2) : percentPositiveOverall = null;
+  hasRecentReviews ? percentPositiveRecent = (initialReviewStats.totalPositiveRecentReviewCount / initialReviewStats.totalRecentReviewCount).toFixed(2) : percentPositiveRecent = null;
```

Comment:
just from a readability standpoint, you can create variables for initialReviewStats.totalPositiveReviewCount, and  initialReviewStats.totalReviewCount because they're so lengthy.